src/main/kotlin/pl/edu/agh/gem/internal/service/FinanceService.kt

```diff
@@ -31,30 +36,39 @@ class FinanceService(
         }
     }
 
-    private fun buildBalancesMap(groupId: String): MutableMap<String, MutableMap<String, BigDecimal>> {
-        val groupData = groupManagerClient.getGroup(groupId)
-        val currencies = groupData.currencies.map { it.code }
-        val memberIds = groupData.members.members.map { it.id }
+    fun blockSettlements(groupId: String, currency: String) {
+        settlementsRepository.blockSettlements(groupId, currency)
+    }
 
-        return currencies.associateWith { _ ->
-            memberIds.associateWith { BigDecimal.ZERO }.toMutableMap()
-        }.toMutableMap()
+    fun saveBalances(balances: Balances) {
+        balancesRepository.save(balances)
     }
 
-    fun getBalances(groupId: String): Balances {
-        val balances = buildBalancesMap(groupId)
+    fun getBalances(groupId: String): List<Balances> {
+        return balancesRepository.getBalances(groupId)
```

Comment:
check if we have all users and all currenceis in the balances list if not we fetch new